### PR TITLE
Use track list as the source of truth about what files are selected, and avoid empty tracks

### PR DESCRIPTION
**Pull request by adamnovak**
This removes the dedicated variables for what files are selected in each dropdown, and instead uses an array of tracks.

It also makes graph/haplotype paths use a `sourceTrackID` like reads do.

It also removes the assumptions that the graph is track 0, the haplotypes if any are track 1, and the reads are tracks 2 and 3. Any file type should be able to be any track number.

This is all in service of making following a copied link actually populate the file dropdowns. When you visit a copied link, it fills in the view target's tracks. Now we use that information to render the dropdowns, so #242 should be fixed.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -14,6 +14,33 @@ import Footer from "./components/Footer";
...
+      if (tracks[key].files[0].type === "read") {
+        schemes[key] = {...config.defaultReadColorPalette};
+      } else {
+        schemes[key] = {...config.defaultHaplotypeColorPalette};
+      }
+    }
+  }
+  
+  return schemes;
+}
```
Comment (ducku) [inline src/App.js:42]:
```suggestion
function addTracksToColorSchemes(tracks, schemes) {

  console.log("Initializing coloring for tracks: ", tracks)

  for (const key in tracks) {
    if (schemes[key] === undefined) {
      // We need to adopt a color scheme
      if (tracks[key].files[0].type === "read") {
        schemes[key] = {...config.defaultReadColorPalette};
      } else {
        schemes[key] = {...config.defaultHaplotypeColorPalette};
      }
    }
  }

  return schemes;
}
```

File: src/App.js [hunk 2]
```diff
@@ -71,9 +94,34 @@ class App extends Component {
...
+        for (const key in newViewTarget.tracks) {
+          if (newColorSchemes[key] === undefined) {
+            // We need to adopt a color scheme
+            if (newViewTarget.tracks[key].files[0].type === "read") {
+              newColorSchemes[key] = {...config.defaultReadColorPalette};
+            } else {
+              newColorSchemes[key] = {...config.defaultHaplotypeColorPalette};
+            }
+          }
+        }
```
Comment (ducku) [inline src/App.js:113]:
```suggestion
        let newColorSchemes = state.visOptions.colorSchemes.slice();
        addTracksToColorSchemes(newViewTarget.tracks, newColorSchemes);

```

File: src/components/TubeMapContainer.js [hunk 3]
```diff
@@ -141,36 +141,52 @@ class TubeMapContainer extends Component {
...
-        // Go through every read and assign it a source file number
-        for (let i = 0; i < readsArr.length; i++) {
-          for (let j = 0; j < readsArr[i].length; j++) {
-            readsArr[i][j].sourceTrackID = sourceTrackIDs[i];
-          }
-        }
+        console.log("Graph track: " + graphTrackID + " Haplotype track: " + haplotypeTrackID);
+
+        const nodes = tubeMap.vgExtractNodes(json.graph);
+        // TODO: For now claim haplotypes came from the graph
```
Comment (ducku) [inline src/components/TubeMapContainer.js:174]: I'm not sure if I understand what needs to be done here. Do haplotypes need to be extracted separately?

File: src/components/HeaderForm.js [hunk 4]
```diff
@@ -169,6 +194,79 @@ class HeaderForm extends Component {
...
+        if (track.files[0].type === type) {
+          console.log("See file " + seenTracksOfType + " of right type");
+          if (seenTracksOfType === index) {
+            if (file !== "none") {
+              // We want to adjust it, so keep a modified copy of it
+              let newTrack = JSON.parse(JSON.stringify(track));
+              newTrack.files[0].name = file;
+              newTracks[key] = newTrack;
+            }
+            // If the file is "none" we drop the track.
```
[the remark this answers is not in the document]
Comment (adamnovak) [reply src/components/HeaderForm.js:221]: Yeah, I think we want to compact them eventually. But I figured starting on that here wouldn't work well with what you are doing for the track picker list, which wants thinks keyed by stable IDs. I think we need to make that change after integrating.

**Pull request merged**